forum/qa-plugin/user-activity-log/user-activity-log.php

```diff
@@ -0,0 +1,79 @@
+<?php
+    if (!defined('QA_VERSION')) {
+        header('Location: ../../');
+        exit;
+    }
+    
+    require_once QA_INCLUDE_DIR.'app/admin.php';
+
+    class user_activity_log 
+    {
+        private $directory;
+        private $urltoroot;
+
+        public function match_request($request) 
+        {
+            if(qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR){
```

Comment:
Skoro w `else` zwracany jest `false`, to całość można zwrócić w postaci ternarki:
```php
return qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR ? $request === 'user-activity-log' : false;
```
Nie wiem jak PHP radzi sobie z obsługą wyrażeń boolean-owych i [ewaluacją short-circuit](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR#short-circuit_evaluation), ale jeśli `qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR` zwraca boolean, to może można całość skrócić do:
```php
return qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR && $request === 'user-activity-log';
```

Reply:
@ScriptyChris zdecydowanie dużo lepszy ten drugi zapis. Jest czytelny a wynik `>=` zawsze da `bool`. Pierwszy zapis powodował jakiś czas temu problemy na różnych maszynach (nie wiem czy już poprawiono ten bug) i jeśli już takie coś stosować to warto tak:
```php
return qa_get_logged_in_level() >= QA_USER_LEVEL_EDITOR ? ($request === 'user-activity-log') : false;
```
Może przy tak małej liczbie leveli to nie ma większego znaczenia, ale lepiej zapobiegać niż leczyć. Dlatego zdecydowanie lepszy ten drugi zapis.